src/compiler/transformers/es2015.ts

```diff
@@ -980,30 +980,32 @@ namespace ts {
             const statements: Statement[] = [];
             resumeLexicalEnvironment();
 
+            // In derived classes, there may be code before the necessary super() call
+            // We'll remove pre-super statements to be tacked on after the rest of the body
+            const existingPrologue = takeWhile(constructor.body.statements, isPrologueDirective);
+            const { superCall, superStatementIndex } = splitConstructorBodyStatementsOnSuper(constructor.body.statements, existingPrologue);
+            const postSuperStatementsStart = superStatementIndex === -1 ? existingPrologue.length : superStatementIndex + 1;
+
             // If a super call has already been synthesized,
             // we're going to assume that we should just transform everything after that.
             // The assumption is that no prior step in the pipeline has added any prologue directives.
-            let statementOffset = 0;
-            if (!hasSynthesizedSuper) statementOffset = factory.copyStandardPrologue(constructor.body.statements, prologue, /*ensureUseStrict*/ false);
+            let statementOffset = postSuperStatementsStart;

```

Comment:
There's a few problems in our RWC suite. The transpilation for initializers is happening *after* the statements that come before `super()`, but directly before `super()`, for example:

```ts
// source
declare class Base {
  constructor(a: any);
}
class Derived extends Base {
  x: boolean;
  constructor(a: any, x = false) {
    if (!a) {
      x = false;
    }
    super(a);
    this.x = x;
  }
}

// prior output (ES5)
var Derived = (function(_super) {
  function Derived(a, x) {
    if (x === void 0) { x = false; } // parameter initializer
    var _this = this;
    if (!a) {
      x = false;
    }
    _this = _super.call(this, a) || this;
    _this.x = x;
  }
})(Base);

// new output (ES5)
var Derived = (function(_super) {
  function Derived(a, x) {
    var _this = this;
    if (!a) {
      x = false;
    }
    if (x === void 0) { x = false; } // parameter initializer (moved)
    _this = _super.call(this, a) || this;
    _this.x = x;
  }
})(Base);
```

The updated output runs the parameter initializers too late. I imagine the same thing is happening for rest parameters. Can you add a test case to verify?

Reply:
Funny enough, `tests/baselines/reference/thisInConstructorParameter2.js` already did verify the default parameter portions; it just got buried among the other changes.